lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2476,10 +2430,87 @@ constructTransaction
             }
   where
     ti :: TimeInterpreter (ExceptT PastHorizonException IO)
-    ti = timeInterpreter (ctx ^. networkLayer)
+    ti = timeInterpreter (apiLayer ^. networkLayer)
 
+    walletId = getApiT apiWalletId
     singleton x = [x]
 
+    validMintingBurning
+        :: ApiConstructTransactionData n
+        -> (SlotNo, SlotNo)
+        -> Either ErrConstructTx (Maybe (NonEmpty (ApiMintBurnData n)))
+    validMintingBurning tx validity = do
```

Comment:
Since `valid` is an adjective, I'm not sure I can understand the behaviour of the function from the name, though it does seem to indicate some kind of validation is taking place?

The type signature indicates that we're returning a `ApiMintBurnData` object.

Looking into the function definition, it seems we're performing some checks, and filling in some missing data (asset names).

Perhaps use the verb `parse` here instead, and reflect the type of object returned in the name?

So:

```suggestion
    parseMintBurnData
        :: ApiConstructTransactionData n
        -> (SlotNo, SlotNo)
        -> Either ErrConstructTx (Maybe (NonEmpty (ApiMintBurnData n)))
    parseMintBurnData tx validity = do
```

Reply:
My idea was to use nouns instead of verbs, e.g. "baked cake is a raw cake after an hour spent in the oven" as opposed to "bake a cake for one hour". If this idea doesn't resonate well with others, I might change it to verbs, but I'd like to at least try.

So this expression `validMintingBurning` represents a "baked cake" :) Even though we already know that [the cake is a lie](https://en.wikipedia.org/wiki/The_cake_is_a_lie).

The bigger picture is that HTTP API is a boundary, to one side of which we have untrusted dirty raw cakes (all kinds of mess the man can invent intentionally or not) and to the other side of it (internal-facing) we have well-formed baked cakes, suitable for passing to an API-unaware wallet logic.

Reply:
I still suspect that using a verb here will be clearer. Anything from the following selection could work, so long as we're consistent:
- validate
- parse
- check
- verify
- extract
- decode
- ...

But given that we're aiming for the concept of "parse, don't validate", I would suggest that "parse" is one of our best choices here.

About the second half of the name:

According to my understanding, for some "thing" to be "valid", that "thing" must usually be noun. And I'm not sure what it means to have a `MintingBurning`. 😺 

However, we do already have a `MintBurnData` type. I would recommend making the name reflect the name of the data type, if possible. (So `parseFoo` parses a value of type `Foo`.)

Reply:
> According to my understanding, for some "thing" to be "valid", that "thing" must usually be a noun. And I'm not sure what it means to have a MintingBurning. smiley_cat

Fair point. `mintingBurning` is an original name. I've renamed it to the `mintBurnData` everywhere.

Reply:
Typically I try to avoid using `Info` and `Data` in names as everything is info and data, but in this case one can interpret it as if its a prefabricated thing that eventually could become a `MintBurn`